enterprise/api_client/enterprise.py

```diff
@@ -0,0 +1,50 @@
+
+# -*- coding: utf-8 -*-
+"""
+Client for communicating with the Enterprise API.
+"""
+from __future__ import absolute_import, unicode_literals
+
+from django.conf import settings
+from django.core.cache import cache
+
+from enterprise.api_client.lms import LmsApiClient
+from enterprise.utils import get_cache_key, traverse_pagination
+
+
+class EnterpriseApiClient(LmsApiClient):
+    """
+    Object builds an API client to make calls to the Enterprise API.
+    """
+
+    API_BASE_URL = settings.LMS_ROOT_URL + 'enterprise/api/v1/'
+
+    def get_all_enterprise_courses(self, enterprise_customer):
+        """
+        Query the Enter API for the course details of the given course_id.
+
+        Arguments:
+            enterprise_customer (Enterprise Customer): Enterprise customer for fetching courses
+
+        Returns:
+            dict: A dictionary containing details about the course, in an enrollment context (allowed modes, etc.)
+        """
+        api_resource_name = 'enterprise-customer'
```

Comment:
@brittneyexline the current code should work as expected. The stacktrace you provided is relatively old code as you can see here:
```python
File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/enterprise/api_client/enterprise.py", line 39, in get_all_enterprise_courses
    endpoint = getattr(self, api_resource_name)
```
the current implementation is:
```python
endpoint = getattr(self.client, api_resource_name)(enterprise_customer.uuid).courses
```
notice the difference `client` vs `self.client`.